src/librustc/middle/ty.rs

```diff
@@ -6112,7 +6086,7 @@ impl<'tcx> ctxt<'tcx> {
     }
 
     // Returns a list of `ClosureUpvar`s for each upvar.
-    pub fn closure_upvars(typer: &Typer<'tcx>,
+    pub fn closure_upvars<'a>(typer: &infer::InferCtxt<'a, 'tcx>,
```

Comment:
Now that this no longer operates on a trait object, it should be moved to the `closure_upvars` method on the typer type (`InferCtxt` in this case).